authenticate/claims.go

```diff
@@ -19,7 +20,14 @@ func (uacClaims *UACClaims) AuthenticatedForInstrument(instrumentName string) bo
 	if strings.EqualFold(uacClaims.UacInfo.InstrumentName, instrumentName) {
 		return true
 	}
-	if uacClaims.UacInfo.InstrumentName == "dia2299a" && instrumentName == "dia2299b" {
+	return uacClaims.CheckDiaInstrument(uacClaims.UacInfo.InstrumentName, instrumentName)
+}
+
+func (uacClaims *UACClaims) CheckDiaInstrument(instrumentName1, instrumentName2 string) bool {
+	diaA, _ := regexp.MatchString(`^dia\d{4}a$`, instrumentName1)
```

Comment:
I'm not a massive fan of using the blank identifier as Go is supposed to force you in to handling your errors. I would write something like:

	diaA, err := regexp.MatchString(`^dia\d{4}a$`, instrumentName1)
	if err != nil {
		fmt.Print(err)
	}
	diaB, err := regexp.MatchString(`^dia\d{4}b$`, instrumentName2)
	if err != nil {
		fmt.Print(err)
	}
	if diaA && diaB {
		return true
	}
	return false
}

So err if handled when it is thrown

Reply:
Chatted this through with James, and added the error code, but let the logic the same